### ISSUE TITLE
Using grpc-web-client with Angular (angular-cli)

### ISSUE THREAD

**Issue by slavede** [improbable-eng/grpc-web#191]
Whenever I tried to serve/build project I get:

ERROR in ./node_modules/grpc-web-client/dist/transports/nodeHttp.js
Module not found: Error: Can't resolve 'http' in '<path_to_project>\node_modules\grpc-web-client\dist\transports'
ERROR in ./node_modules/grpc-web-client/dist/transports/nodeHttp.js
Module not found: Error: Can't resolve 'https' in '<path_to_project>\node_modules\grpc-web-client\dist\transports'

Anyone encountered it ever?

It seems that require('http') from nodeHttp.js can't hit NodeJS's http...

**Comment by Drogglbecher** [comment]
I have the same problem since updated to Angular 6. Could be a problem with the HttpModule Deprecation but not sure at all atm.

**Comment by MHDante** [comment]
Yep. Same problem on Angular 6.

**Comment by dirkluijk** [comment]
Any workaround for this?

**Comment by nekrondev** [comment]
Dear all,

I've got another problem besides the nodeHttp.js missing issue and kindly ask for your help.

I generated sample typescript code for my ng6 project located at "./src/app/generated/*". Now when I try to build or serve the project I get the following error:
```
ERROR in ./src/app/generated/proto/sweeguiapi_pb_service.js
Module not found: Error: Can't resolve '../proto/sweeguiapi_pb' in 'C:\src\go\src\swee_restart\sweefrontend\src\app\generated\proto'
```
I also imported the stubs from a service that is located at "./src/app":

```
import { GuiApiService } from './generated/proto/sweeguiapi_pb_service';
import { GetClusterReq, GetClusterRes, STATE } from './generated/proto/sweeguiapi_pb';
```
which gives me:

```
ERROR in ./src/app/monitor.service.ts
Module not found: Error: Can't resolve './generated/proto/sweeguiapi_pb' in 'C:\src\go\src\swee_restart\sweefrontend\src\app'
```
VSCode however does not show and missing modules and IDE works as expected. 

Whats wrong with my setup?

I am using the following deps:

```
 "dependencies": {
    "@angular/animations": "^6.0.0",
    "@angular/cdk": "^6.0.1",
    "@angular/common": "^6.0.0",
    "@angular/compiler": "^6.0.0",
    "@angular/core": "^6.0.0",
    "@angular/forms": "^6.0.0",
    "@angular/http": "^6.0.0",
    "@angular/material": "^6.0.1",
    "@angular/platform-browser": "^6.0.0",
    "@angular/platform-browser-dynamic": "^6.0.0",
    "@angular/router": "^6.0.0",
    "@types/google-protobuf": "^3.2.7",
    "angular-in-memory-web-api": "^0.6.0",
    "core-js": "^2.5.4",
    "google-protobuf": "^3.5.0",
    "grpc-web-client": "^0.6.2",
    "hammerjs": "^2.0.8",
    "npm": "^6.0.0",
    "rxjs": "^6.0.0",
    "ts-protoc-gen": "^0.6.0",
    "zone.js": "^0.8.26"
  },
```
Thanks for your help!

**Comment by MHDante** [comment]
@MarcusLongmuir This change seems related to this post: https://github.com/angular/angular-cli/issues/9827#issuecomment-369578814

What google is saying is that the browser version of grpc-web should not rely on grpc libraries

**Comment by MHDante** [comment]
To anyone still fighting with this, here's a workaround. (For frontend only)

In your tsconfig.json, add:

``` javascript
{
"compilerOptions": {
    "paths": {
      "https": [ "app/patch.js" ],
      "http": [ "app/patch.js" ],
    }
}
```
and in your app folder, create a new file `patch.js` with the following content:

``` javascript
//https://github.com/improbable-eng/grpc-web/issues/191
```

**Comment by nekrondev** [comment]
@MHDante: Workaround works as expected. 

Meanwhile I found out what my problem was. I used the protoc command from the examples and generated the js_out into a different directory (binary) so that type definition and js file where located in two different places making it unable to resolve by ng build. 

Well, it must be the 30 degrees Celsius over here in Germany that toasted my brain.

**Comment by curiouscod3** [comment]
It doesn't work dude @MHDante

**Comment by MHDante** [comment]
The exact path to patch.ts is dependent on how you've set up your project.

**Comment by Diemauerdk** [comment]
I have this problem too, really hope for a solution soon :(

**Comment by fengyue** [comment]
The "compilerOptions" workaround doesn't work.  @MHDante any ideas?

My config is
"paths": {
      "https": [ "./src/app/patch.ts" ],
      "http": [ "./src/app/patch.ts" ],
}

**Comment by johanbrandhorst** [comment]
Would this be fixed by #204? Sounds like we could hit 2 birds with one stone.

**Comment by kmturley** [comment]
I'm working on another Angular gRPC example project here:
https://github.com/kmturley/angular-nest-grpc
